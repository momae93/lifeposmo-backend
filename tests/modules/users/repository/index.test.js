const buildUserRepository = require('../../../../src/modules/users/repository/');

describe('[MODULES][USERS][REPOSITORY] : user repository', () => {
  it('should build user repository with no db client passed', async () => {
    // LOCAL MOCKS
    const mockDbClient = null;

    // FUNCTION TESTED
    const userRepository = buildUserRepository(mockDbClient);

    // EXPECTED
    const expectedRepository = {
      createUser: expect.any(Function),
      deleteUser: expect.any(Function),
      getAllUsers: expect.any(Function),
      getUserById: expect.any(Function),
    };

    expect(userRepository).toMatchObject(expectedRepository);
  });
});
