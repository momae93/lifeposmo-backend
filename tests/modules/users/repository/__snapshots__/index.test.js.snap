// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MODULES][USERS][REPOSITORY] : user repository should build user repository with no db client passed 1`] = `
Object {
  "createFavoriteUser": [Function],
  "createUser": [Function],
  "deleteFavoriteUser": [Function],
  "deleteUser": [Function],
  "getAllUsers": [Function],
  "getFollowersByIdUser": [Function],
  "getFollowingByIdUser": [Function],
  "getUserById": [Function],
}
`;
